rdf4j-java/src/main/java/eu/neverblink/jelly/convert/rdf4j/rio/JellyParserFactory.java

```diff
@@ -0,0 +1,22 @@
+package eu.neverblink.jelly.convert.rdf4j.rio;
+
+import static eu.neverblink.jelly.convert.rdf4j.rio.JellyConstants.JELLY_RDF_FORMAT;
+
+import eu.neverblink.jelly.convert.rdf4j.Rdf4jConverterFactory;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.RDFParser;
+import org.eclipse.rdf4j.rio.RDFParserFactory;
+
+public final class JellyParserFactory implements RDFParserFactory {
+
+    @Override
+    public RDFFormat getRDFFormat() {
+        return JELLY_RDF_FORMAT;
+    }
+
+    @Override
+    public RDFParser getParser() {
+        final var converterFactory = new Rdf4jConverterFactory();
```

Comment:
It makes no sense to build a new factory on each call.

For: `rdf4j`, `jena`, `rdf4jPatch`, `jenaPatch` please make these factories into singletons, with a static member INSTANCE or a static method getInstance (whatever you prefer) and make the constructor private. There is absolutely zero use in having more than one instance on the heap.